admin/category_product_listing.php

```diff
@@ -1264,7 +1264,8 @@
 
               if (empty($product_types)) {
                 // There are no restricted product types so offer all types instead
-                $sql = "SELECT * FROM " . TABLE_PRODUCT_TYPES;
+                $lang_suffix = (!empty($_SESSION['languages_code']) && $_SESSION['languages_code'] != 'en') ? '_' . $_SESSION['languages_code'] : '';
+                $sql = "SELECT type_id, type_name" . $lang_suffix . " AS type_name, type_handler, type_master_type, allow_add_to_cart, default_image, date_added, last_modified FROM " . TABLE_PRODUCT_TYPES;
```

Comment:
I think you can continue to use `SELECT *`, without having to list all column names.
I'm pretty sure that a later-declared column replaces the former.
So `SELECT *, type_name$lang_suffix AS type_name` would probably work fine. Have to test, to be sure it doesn't come back as `type_name(1)`.

Reply:
Edit: actually, in this case most of the fields aren't needed. It's only using the `type_id` and `type_name` from this query anyway, because it's only populating a dropdown.

Reply:
OK, I am going to test that and see if `type_id` and `type_name` are enough.

Reply:
`type_id` and `type_name` are definitely enough, but just for information, I although tested `SELECT *, type_name$lang_suffix AS type_name` and it worked fine. The last declaration takes over.
I will update that query.

Reply:
I took time to although check other queries and optimized them.